packages/runtime-common/search-index.ts

```diff
@@ -635,6 +639,69 @@ export class SearchIndex {
         return null;
       });
   }
+
+  private async buildContainsMatchers(
+    filterValue: ContainsFilter['contains'],
+    ref: CardRef
+  ): Promise<(entry: SearchEntry) => boolean | null> {
+    // TODO when we are ready to execute queries within computeds, we'll need to
+    // use the loader instance from current-run and not the global loader, as
+    // the card definitions may have changed in the current-run loader
+    let api = await this.loadAPI();
+
+    let matchers: ((instanceData: Record<string, any>) => boolean | null)[] =
+      [];
+
+    for (let [name, value] of Object.entries(filterValue)) {
+      // Load the stack of fields we're accessing
+      let fields: Field[] = [];
+      let nextRef: CardRef | undefined = ref;
+      let segments = name.split('.');
+      while (segments.length > 0) {
+        let fieldName = segments.shift()!;
+        let field = await this.loadField(nextRef, fieldName);
+        fields.push(field);
+        nextRef = identifyCard(field.card);
+        if (!nextRef) {
+          throw new Error(`could not identify card for field ${fieldName}`);
+        }
+      }
+
+      let queryValue = api.getQueryableValue(fields[fields.length - 1], value);
+      let matcher = (instanceValue: any) => {
+        if (instanceValue === undefined && queryValue != null) {
+          return null;
+        }
+        // allows queries for null to work
+        if (queryValue == null && instanceValue == null) {
+          return true;
+        }
+        return (instanceValue as string).toLowerCase().includes(queryValue.toLowerCase());
+      };
+      while (fields.length > 0) {
+        let nextField = fields.pop()!;
+        let nextMatcher = nextField.queryMatcher(matcher);
+        matcher = (instanceValue: any) => {
+          if (instanceValue == null && queryValue != null) {
+            return null;
+          }
+          if (instanceValue == null && queryValue == null) {
+            return true;
+          }
```

Comment:
please include tests for these situations. (but also i'm still confused what a queryValue of `null` means)